source/retryable-writes/retryable-writes.rst

```diff
@@ -511,6 +513,15 @@ experience across all drivers.
 Why expose users to server-side errors for unsupported operations?
 ------------------------------------------------------------------
 
+.. todo:: Determine how this should change. Do we end up using whitelists for
+   supported operations (based on wire protocol verison) and avoid injecting
+   transaction IDs on all write ops, or rely on the server to raise an error if
+   unsupported ops would be retried? In both cases, it's possible for users to
+   be unaware that some operations are not supported until some error in prod.
+   Given the MongoClient-wide scope of retryWrites, we likely don't want to
+   prohibit unsupported operations outright (either in the client API or by
+   having the server raise an exception on the first attempt).
+
```

Comment:
I guess we have to whitelist supported operations and only include txnNum if the command is in the whitelist. Document carefully in the MongoDB Manual which operations are retryable in which server versions. The alternative, which is to require that users only execute retryable operations or else receive an error, might make retryable writes too hard to use in applications with a lot of existing code. For future work we can specify a format in the isMaster reply that communicates the server's list of retryable write operations to the driver, which will relieve us of the burden, but for 3.6 we can hardcode the whitelist in all drivers.

If we follow this approach, I propose we allow unacknowledged writes with a retryWrites client. They're not retryable, obviously, but we should allow them for consistency with other non-retryable operations.

I am very ambivalent about this opinion, however.